### ISSUE TITLE
Issues with auto-import react-typescript project in tsx files

### ISSUE THREAD

**Issue by JustFly1984** [codota/tabnine-vscode#113]
**please complete the following information:**	
 - OS version: Mac Catalina	
 - Editor version 1.46.0
 - Programming language: TypeScript
 - TabNine extension version:
 
**Issue Details:**
In my React files I have on top of the file
```
import * as React from 'react'
```

and it is correct Typescript import statement considering tsconfig.json settings
```
{
  "esModuleInterop": false,
  "allowSyntheticDefaultImports": false
}
```

The issue is that tabnine adds next lines repeatedly, sometimes multiple time while I'm editing components.
```
const React = require('react')
```
after last import statement.

We are using only ES6 import statements around the codebase, also some import() for async code splitting.

I guess tabnine-vscode looks for `import React from 'react'` and finds none, but in our case it is incorrect import.

We would not change typescript settings, cos it breaks importing some of dependencies we have.

Also I have noticed that tabnine-vscode often imports same stuff multiple times, even in destructured import statements

example of weird behavior:

```
import { useSelector, shallowEqual, useDispatch,  useDispatch,  useDispatch,  useDispatch, useDispatch, } from 'react-redux'
```

One more issue is then I'm creating new interface, after typing `interface Props {` I press enter to jump on next line,but instead of that I get `interface Props { ReduxState}` on single line. I do not see any autocomplete suggestion before I press enter.

I know my codebase, and `ReduxState` is a type of redux store, but how is it related in the context of typing interface declaration...

**Comment by dimacodota** [comment]
Hey @JustFly1984, 

can you please provide us the following:
 -  turn off TabNine's auto import and try to reproduce it again
    go to settings -> Tabnine: Experimental Auto Imports and set it to false.

 -  a list of your installed extensions
   `code --list-extensions`

**Comment by JustFly1984** [comment]
@dimacodota

It is not reproducible without tabnine experiment import. I guess all of that is bugs in experimental imports feature.

list of extensions:

```
aaron-bond.better-comments
akamud.vscode-theme-onedark
anseki.vscode-color
antfu.i18n-ally
awwit.typed-css-modules-plugin
bungcip.better-toml
christian-kohler.npm-intellisense
christian-kohler.path-intellisense
clinyong.vscode-css-modules
codezombiech.gitignore
CoenraadS.bracket-pair-colorizer
DavidAnson.vscode-markdownlint
dbaeumer.vscode-eslint
deepcode.deepcode
DigitalBrainstem.javascript-ejs-support
eamodio.gitlens
EditorConfig.EditorConfig
eg2.vscode-npm-script
esbenp.prettier-vscode
etaiklein.formatted-message
fabiospampinato.vscode-open-in-github
formulahendry.auto-close-tag
formulahendry.auto-rename-tag
formulahendry.terminal
funkyremi.vscode-google-translate
hedinne.stylelint-disable-snippets
HookyQR.minify
kiteco.kite
kumar-harsh.graphql-for-vscode
lannonbr.vscode-js-annotations
mhutchie.git-graph
michelemelluso.gitignore
mikestead.dotenv
msjsdiag.debugger-for-chrome
naumovs.color-highlight
oderwat.indent-rainbow
redhat.vscode-yaml
snyk-security.vscode-vuln-cost
steoates.autoimport
streetsidesoftware.code-spell-checker
streetsidesoftware.code-spell-checker-russian
stylelint.vscode-stylelint
TabNine.tabnine-vscode
Tyriar.terminal-tabs
waderyan.gitblame
```

**Comment by dimacodota** [comment]
@JustFly1984 ,
can you try to disable steoates.autoimport ?
TabNine auto import will invoke the default "code action" that available at specific location as same as the lightbulb(ctrl + space), other extensions may register additional code actions.

**Comment by JustFly1984** [comment]
@dimacodota `steoates.autoimport` already disabled, just in the list.

**Comment by JustFly1984** [comment]
Is there any update for auto import?

**Comment by boaz-codota** [comment]
Hey @JustFly1984 I've been trying to tackle this, and I suspect this relates to a TypeScript's resolved issue: 
https://github.com/microsoft/TypeScript/pull/37027

Could you report on the version of TypeScript you're using, and possibly try to update it, to see if it is possibly fixed?

**Comment by Silventino** [comment]
Same issue here. 
Reproduced with the latest release of Typescript (3.9.7)
Disabled TabNine for now...

**Comment by dimacodota** [comment]
the duplicate imports bug was fixed in 3.0.0 version
